src/site/content/algolia.11tydata.js

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const removeMarkdown = require('remove-markdown');
+const {livePosts} = require('../_filters/live-posts');
+const {feed: authorsFeed} = require('../_collections/hooks/authors');
+const {feed: tagsFeed} = require('../_collections/hooks/tags');
+
+/**
+ * Shrink the size of the given fulltext to fit within a certain limit, at the
+ * nearest found newline character.
+ *
+ * @param {string} fulltext
+ * @param {number=} limit
+ * @return {string}
+ */
+function limitText(fulltext, limit = 7500) {
+  if (fulltext.length <= limit) {
+    return fulltext;
+  }
+
+  // Find the nearest prior newline to the 10k limit.
+  let newlineIndex = fulltext.lastIndexOf('\n', limit);
+  if (newlineIndex === -1) {
+    newlineIndex = limit;
+  }
+  return fulltext.slice(0, newlineIndex);
+}
+
+module.exports = () => {
+  return {
+    eleventyComputed: {
+      algolia: ({collections}) => {
+        if (process.env.ELEVENTY_ENV !== 'prod') {
+          return [];
+        }
+        // For now, hard-code language to English.
+        const lang = 'en';
+        const validTags = ['post'];
+        /** @type EleventyCollectionItem[] */
+        const eleventyPosts = collections.all
+          .filter((item) => {
+            // nb. There's no easy 'getFilteredByMultipleTag' method in Eleventy.
+            if (!Array.isArray(item.data.tags)) {
+              return false;
+            }
+            return item.data.tags.some((tag) => validTags.includes(tag));
```

Comment:
Since there's only one tag could you instead start with `collections.post` instead of `collections.all`? Eleventy will have already filtered this collection for us.